### PR TITLE
upgrade react to v18 (#173)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- upgrade test libraries
- install without optional dependencies (npm install --omit=optional) to prevent wront version of react types to be installed
- install @types/react
- specify missing types
- remove deprecation warnings

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
all works just fine. well done Fred!

File: gui/src/components/Taipy/DateSelector.spec.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React from "react";
-import { render, fireEvent } from "@testing-library/react";
+import { render } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 import "@testing-library/jest-dom";
 
-import { LocalizationProvider } from "@mui/lab";
+import { LocalizationProvider } from "@mui/x-date-pickers";
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/DateSelector.spec.tsx:6]: Hmmm... Introduction of Mui X...

File: setup.py [hunk 2]
```diff
@@ -50,7 +50,7 @@
 def _build_webapp():
     already_exists = Path(f"./taipy/gui/webapp/index.html").exists()
     if not already_exists:
-        os.system("cd gui && npm ci && npm run build")
+        os.system("cd gui && npm ci --omit=optional && npm run build")
```
Comment (FabienLelaquais) [inline setup.py:53]: Shouldn't there be a --only=prod at this point?

**Review by FabienLelaquais — APPROVED** [review]
Great, great job!

File: setup.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply setup.py:53]: --only=prod is not supported since at least npm 6.3

File: gui/src/components/Taipy/DateSelector.spec.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/DateSelector.spec.tsx:6]: it replaces @mui/lab for date pickers and I'd say it is more stable than lab

**Pull request merged**